src/Zenova/Platform/Windows.cpp

```
@@ -87,6 +89,97 @@ namespace Zenova::PlatformImpl {
 			FreeLibraryAndExitThread(reinterpret_cast<HMODULE>(CleanupVariables), 0);
 		}
 	}
+
+	void* LoadModModuleAndResolveImports(const std::string& module)
+	{
+		void* hModule = LoadLibraryExA(module.c_str(), NULL, DONT_RESOLVE_DLL_REFERENCES);
+		ResolveModModuleImports(hModule, module);
+		return hModule;
+	}
+
+	typedef BOOL(APIENTRY* DllMainFunction)(HMODULE, DWORD, LPVOID);
+	void ResolveModModuleImports(void* hModule, const std::string& moduleName)
+	{
+		if (!hModule) {
+			throw std::invalid_argument(fmt::format("[{}] Could not resolve the imports for the module because it was nullptr.", __FUNCTION__));
```

Can we avoid using exceptions? We should only be logging it and silently failing to load at best (same with the rest of this function)